mitodl/open-discussions
# Index user profiles in Elasticsearch

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #1317

#### What's this PR do?
- Adds a profile index
- Adds a profile mapping and refactors common mapping fields
- Index new profiles
- Update the profile document when a profile is changed.
- Update all relevant post/comment docs when a profile name or avatar is changed.
- Corrects a mismatch between channel name and title when saving `Channel` objects.

#### How should this be manually tested?
- Run `python manage.py recreate_index`.  When done, there should be a `discussions_local_profile_default` index with documents for all existing profiles.
- Edit your profile, changing bio and headline.  The changes should be reflected in the index doc for your profile.
- Change the name and avatar of your profile.  The `author_name` and `author_avatar_small` fields of all your posts and comments should also be changed in ES.
- Register a new user by email, and another new user via Micromasters.  When complete, the ES doc for each profile should exist and have the correct info.
